nslsii/temperature_controllers.py

```diff
@@ -68,13 +85,13 @@ def status_indicator(value, timestamp, **kwargs):
         self.setpoint.put(set_value)
 
         # subscribe to the read value to indicate the set is done.
-        self.readback.subscribe(status_indicator)
+        self._cid = self.readback.subscribe(status_indicator)
 
         # hand the status object back to the RE
         return status
 
     def stop(self):
         # overide the lock on any in progress sets
-        self._set_lock = False
+        self._set_lock.release()
```

Comment:
Should this cancel the timer and clear the subscription?

Reply:
this has been fixed